### PR TITLE
add helpers and types for interacting with stardog connection cookie

### PR DESCRIPTION
**Pull request by SpiralP**
also moved files around; lemme know if this layout works!

### CODE REVIEW

File: src/cookies/analytics.ts [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
     const decoded = JSON.parse(atob(rawCookie));
+
     const consented: boolean = decoded.consented;
-    const identity: string = decoded.identity || '';
-    if (typeof identity !== 'string' || typeof consented !== 'boolean') {
+    const identity: string = decoded.identity;
+
+    if (
+      typeof consented !== 'boolean' ||
+      (typeof identity !== 'undefined' && typeof identity !== 'string')
```
Comment (rmyers) [inline src/cookies/analytics.ts:22]: Maybe we can make this cleaner with a check if it is set first like:

```
const optionalTypeCheck(val: any, type: string): bool => {
   return typeof val !== 'undefined' && typeof val !== type;
}
```

Comment (SpiralP) [reply src/cookies/analytics.ts]: how's this
[`0e8ef5c` (#7)](https://github.com/stardog-union/portal-sdk/pull/7/commits/0e8ef5cd4a14e858b40c884d1442065a99a430ca)

File: src/cookies/connection.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+export const decodeConnectionCookie = (
+  rawCookie: string
+): ConnectionCookie | null => {
+  try {
+    const decoded = JSON.parse(atob(rawCookie));
+
+    const username: string = decoded.username;
+    const password: string = decoded.password;
```
[the remark this answers is not in the document]
Comment (SpiralP) [reply src/cookies/connection.ts]: yea good point
[`a7ce28b` (#7)](https://github.com/stardog-union/portal-sdk/pull/7/commits/a7ce28b687c16e694aa6397a619be4d8f9e3f204)

File: src/cookies/connection.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      endpoint,
+      tokenEndpoint,
+      graphqlEndpoint,
+    };
+  } catch {
+    return null;
+  }
+};
+
+export const getConnectionConsentCookie = (): ConnectionCookie | null => {
```
Comment (rmyers) [inline src/cookies/connection.ts:46]: `getConnectionCookie`

File: src/cookies/connection.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  const rawConsentCookie = getCookieValue(CONNECTION_COOKIE_NAME);
```
Comment (rmyers) [inline src/cookies/connection.ts:47]: `rawConnectionCookie`

**Review by rmyers — APPROVED** [review]
Couple nits in the names but looks good

File: src/cookies/connection.ts (hunk 4 above)
Comment (SpiralP) [reply src/cookies/connection.ts]: [`ee12bb8` (#7)](https://github.com/stardog-union/portal-sdk/pull/7/commits/ee12bb86b963dec8e0d63bf61d34b31465d87cde)

File: src/cookies/connection.ts (hunk 3 above)
Comment (SpiralP) [reply src/cookies/connection.ts]: [`ee12bb8` (#7)](https://github.com/stardog-union/portal-sdk/pull/7/commits/ee12bb86b963dec8e0d63bf61d34b31465d87cde)

**Pull request merged**